### PR TITLE
pinkerton: Support pulling images from a TUF repository

### PR DESCRIPTION
**Pull request by lmars**
Image URIs must now contain the repo name in the `name` parameter (e.g. `https://registry.hub.docker.com?name=ubuntu&tag=latest`)

Pulling from a TUF repository requires an initialized TUF file (e.g. by running `tuf-cli init --store /path/to/store.db /path/to/root-keys.json`), but other than that looks similar to pulling a Docker image:

```
$ pinkerton pull --tuf-db /tmp/tuf.db https://s3.amazonaws.com/flynn-test/images?name=flynn/etcd&tag=latest
511136ea3c5a64f264b78b5433614aec563103b4d4702f3ba7d4d2698e22c158 exists
a643fa1f27423cb7d426d52aed06d846fd6a4518b44da74ab03fc78cbfae6a04 exists
0dc93e107f111bf2336516d171ac2b4ea1cd56b9cd2686d773b8208f1e265b3a exists
184af8860f22e7a87f1416bb12a32b20d0d2c142f719653d87809a6122b04663 exists
0ae33c20b11bc13fc2f8629a4a444fb4e1d5a75cfbc52382e67fbcd00cfe832d exists
11a5bb79c9f90a9ce67b410f23907d674063bd8c14abaa8e1f312726ddcdb5aa downloaded
```

Incremental patch for #617.

### CODE REVIEW

**Comment by titanous** [comment]
I think we should vendor Godeps on the go-tuf side before merging this. Also the godeps should be in a separate commit.

File: pinkerton/registry/tuf.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	return os.Remove(t.Name())
+}
+
+func (s *tufSession) get(name string, out interface{}) (io.ReadCloser, error) {
+	tmp, err := ioutil.TempFile("", "pinkerton")
+	if err != nil {
+		return nil, err
+	}
+	name = path.Join("v1", name)
+	if err := s.client.Download(name, &tmpFile{tmp}); err != nil {
```
Comment (titanous) [inline pinkerton/registry/tuf.go]: We should wrap `tmpFile.Close` so that it removes the file, and return the wrapped file.

**Comment by lmars** [comment]
@titanous comments addressed

File: pinkerton/registry/tuf.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	_, err := s.get(fmt.Sprintf("/repositories/%s/tags", s.ref.repo), &tags)
+	return tags, err
+}
+
+type tmpFile struct {
+	*os.File
+}
+
+func (t *tmpFile) Delete() error {
+	t.Close()
```
Comment (titanous) [inline pinkerton/registry/tuf.go]: This needs to be `t.File.Close()` otherwise it will infinite loop.

**Comment by lmars** [comment]
@titanous comments addressed

**Comment by titanous** [comment]
LGTM

**Pull request merged**